src/tests/conformance/messages.rs

```diff
@@ -378,6 +378,116 @@ async fn correctly_lists_peers() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn get_blocks() {
+    // ZG-CONFORMANCE-015
+    //
+    // The node responds to `GetBlocks` requests with a list of blocks based on the provided range.
+    //
+    // We test the following conditions:
+    //  1. unlimited queries i.e. stop_hash = 0
+    //  2. range queries i.e. stop_hash = i
+    //  3. a forked chain (we submit a valid hash, followed by incorrect hashes)
+    //
+    // Test procedure:
+    //  1. Create a node and seed it with the testnet chain
+    //  2. Establish a peer node
+    //  3. For each test case:
+    //      a) send GetBlocks
+    //      b) receive Inv
+    //      c) assert Inv received matches expectations
+    //
+    // The test currently fails for both Zebra and zcashd.
+    //
+    // Current behaviour:
+    //
+    //  zcashd: Fails because it ignores requests when the last hash is the last hash it knows of. We expect it to return
+    //          an empty Inv.
+    //
+    //  zebra: does not support seeding as yet, and therefore cannot perform this test.
+    //
+    // Note: zcashd also excludes the stop_hash from the range, whereas the spec states that it should be inclusive.
+    //       We are taking current behaviour as correct.
+
+    let (zig, node_meta) = read_config_file();
+
+    // Create a node with knowledge of the initial three testnet blocks
+    let mut node = Node::new(node_meta);
+    node.initial_action(Action::SeedWithTestnetBlocks {
+        socket_addr: zig.new_local_addr(),
+        block_count: 3,
+    })
+    .log_to_stdout(true)
+    .start()
+    .await;
+
+    let blocks = Block::initial_testnet_blocks();
+
+    let mut stream = initiate_handshake(node.addr()).await.unwrap();
+    let filter = MessageFilter::with_all_auto_reply();
+
+    // Test GetBlocks(i) -> Inv(i+1..)
```

Comment:
This makes sense to an informed reader but it would be good to have some more context in plain english?

Reply:
Good point